sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/functions.scala

```diff
@@ -524,6 +525,116 @@ case class Sum(child: Expression) extends DeclarativeAggregate {
   override val evaluateExpression = Cast(currentSum, resultType)
 }
 
+case class Corr(
```

Comment:
* Please add ScalaDoc and document the behavior for `null` and `NaN` values.
* Provide a link to the wikipedia page that contains the update formula.